### PR TITLE
[3825][16.0][ADD] stock_picking_consumable_accounting_date

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3825](https://www.quartile.co/web#id=3825&cids=3&model=project.task&view_type=form)

### CODE REVIEW

File: stock_picking_consumable_effective_date/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    show_consumable_date = fields.Boolean(compute="_compute_show_consumable_date")
+
+    def _compute_show_consumable_date(self):
+        for pick in self:
+            pick.show_consumable_date = False
+            if pick.picking_type_code not in ("incoming", "outgoing"):
+                continue
+            if pick.move_ids.with_company(pick.company_id).product_id.filtered(
+                lambda x: x.detailed_type == "consu"
+            ):
```
Comment (yostashiro) [inline stock_picking_consumable_effective_date/models/stock_picking.py]: We do not want to show both "Accounting Date" and "Consumable Effective Date" when a picking contains stockable and consumable products.

```suggestion
            moves = pick.move_ids
            # When all moves are consumables.
            if moves and not moves.with_company(pick.company_id).product_id.filtered(
                lambda x: x.detailed_type == "product"
            ):
```

File: stock_picking_consumable_effective_date/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This module adds a computed field show_consumable_date to stock.picking.
+When there is consumable product in picking, the accounting_date field
+will display with the label "Consumable Effective Date".
```
Comment (yostashiro) [inline stock_picking_consumable_effective_date/readme/DESCRIPTION.rst]: @rinaldifirdaus Please add the business context.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Please evaluate my suggestions.

File: inventory_reporting/reports/inventory_report.py [hunk 3]
```diff
@@ -227,12 +227,20 @@ def generate_storable_report(self, workbook, wizard):
...
+            JOIN product_product prod ON svl.product_id = prod.id
+            JOIN product_template tmpl ON prod.product_tmpl_id = tmpl.id
+            WHERE COALESCE(pick.accounting_date, move.date) BETWEEN %s AND %s
+            AND prod.active = TRUE
+            AND tmpl.detailed_type != 'product'
+        """
+        self.env.cr.execute(query, (wizard.date_start, wizard.date_end))
+        svl_ids = [row[0] for row in self.env.cr.fetchall()]
+
+        base_consu_domain = [("id", "in", svl_ids)]
```
Comment (yostashiro) [inline inventory_reporting/reports/inventory_report.py]: I thought I posted a rather lengthy comment on this in the last review but I guess I forgot to post it.  I am not comfortable with this change and we should consider adding the `accounting_date` field in stock.move (in stock_picking_accounting_date, and inherit and adjust the relevant method in stock_picking_consumable_effective_date) and use this field in `base_consu_domain` (consider consolidating the common parts of `base_consu_domain` and `base_product_domain` into one as appropriate.  Please identify the existing logic that needs to be adjusted and follow up.

Comment (rinaldifirdaus) [reply inventory_reporting/reports/inventory_report.py]: Hi @AungKoKoLin1997 , can you please adjust this line as well in for the consumable report.
I think this field should reflect the date that set in the consumable effective date field or the one from the effective date if consumable effective date is none.
https://github.com/qrtl/axls-custom/blob/d536468ca20116ef62ea1151b7215a91e64e411c/inventory_reporting/reports/inventory_report.py#L297

Comment (AungKoKoLin1997) [reply inventory_reporting/reports/inventory_report.py]: My mistake. I didn't notice we have https://github.com/qrtl/axls-custom/tree/16.0/stock_move_accounting_date. I will adjust depends on this module.

**Comment by rinaldifirdaus** [comment]
@yostashiro @AungKoKoLin1997 

based on this morning discussion, we are going to make adjustment as follows:

- Combine stock_move_accounting_date with stock_picking_accounting_date.

- Need to adjust _compute_accounting_date(stock_move.py line 18) of the stock_move_accounting_date to consider the user's timezone.

- Need to adjust _prepare_account_move_vals of the stock_picking_accounting_date (stock_move.py line 29~) to assign the date of journal entry base on stock move's accounting date not picking's accounting date.

- Use SVL "accounting_date" instead of stock_move_id.date to filter the date start and date end (get_base_domain). consumable products doesn't create a journal entry, to handle that need to adjust the _compute_accounting_date function of the stock_valuation_layer_accounting_date by add condition if there's no related journal entry, rec.accounting_date filled by stock move's accounting date.

File: inventory_reporting/reports/inventory_report.py [hunk 4]
```diff
@@ -227,12 +232,16 @@ def generate_storable_report(self, workbook, wizard):
...
-            ("product_id.detailed_type", "!=", "product"),
-        ]
+        base_domain = self.get_base_domain(wizard)
+        base_consu_domain = expression.AND(
+            [
+                base_domain,
+                [
+                    ("product_id.detailed_type", "!=", "product"),
+                ],
+            ]
```
Comment (yostashiro) [inline inventory_reporting/reports/inventory_report.py]:
```suggestion
            [base_domain, [("product_id.detailed_type", "!=", "product")]]
```

**Review by yostashiro — APPROVED** [review]
Code review.

**Pull request merged**